### PR TITLE
Remove internal args to on & remove methods

### PR DESCRIPTION
**Pull request by mzgol**
Fixes gh-2301

+24 bytes, most of which come from event changes. The manipulation change adds just 5 bytes.

I've also tried to move `on` to a proper `/var/` directory but it only increased the size further.

Any ideas on how to reduce the size increase?

### CODE REVIEW

**Comment by timmywil** [comment]
Nice :+1:

**Comment by timmywil** [comment]
I wonder if it would gzip better if we did `on.call(this, ...`

**Comment by mzgol** [comment]
> I wonder if it would gzip better if we did on.call(this, ...

In `manipulation` switching to `.call` adds 11 bytes, in `event` it adds 3 bytes. I tried that at the beginning but I discovered it's actually larger.

**Comment by dmethvin** [comment]
:+1: thanks for doing this.

**Comment by mzgol** [comment]
I'll land this shortly barring any new remarks.

File: src/manipulation.js [hunk 1]
```diff
@@ -420,7 +424,7 @@ jQuery.fn.extend({
 	},
 
 	detach: function( selector ) {
-		return this.remove( selector, true );
+		return remove( this, selector, true );
```
Comment (gibson042) [inline src/manipulation.js]: There's small savings (4 btyes min+gzip) to defining `detach` immediately before `remove` and moving both of them to the top of the `jQuery.fn.extend` block, but that's about it. I'm in favor of this.

Comment (mzgol) [inline src/manipulation.js]: Currently it's only -3 bytes but I'll do it.

**Pull request merged**